agent/stats/container.go

```diff
@@ -183,3 +201,44 @@ func (container *StatsContainer) terminal() (bool, error) {
 	}
 	return dockerContainer.Container.KnownTerminal(), nil
 }
+
+// syncContainerRestartAggregationData updates a container's restart aggregation data if a container restart has been
+// detected. This method returns true if a restart was detected and returns false otherwise.
+func (container *StatsContainer) syncContainerRestartAggregationData() bool {
+	_, dockerContainerData := container.client.DescribeContainer(container.ctx, container.containerMetadata.DockerID)
+	restartDetected := dockerContainerData.StartedAt.Sub(container.containerMetadata.StartedAt) > 0
```

Comment:
instead of calling the docker API on every stat, could we use the restart tracker here to determine if a restart happened? like 

```
apicontainer.RestartTracker.GetRestartCount() > 0
```

Reply:
I don't think we can do this because `apicontainer.RestartTracker.GetRestartCount() > 0` tells us if a container restart has ever happened before but not if a restart has just happened.

Reply:
I don't think we can do this because `apicontainer.RestartTracker.GetRestartCount() > 0` tells us if a container restart has ever happened before but not if a restart has just happened (which is what we are interested in here).